core/BoidsGenerator.cpp

```diff
@@ -39,5 +59,54 @@ size_t Core::BoidsGenerator::getVerticesBufferSize()
 void Core::BoidsGenerator::updateBoids()
 {
     // Where you need to define your physics function with the three Boids rules;
-    generateBoids();
+    //generateBoids();
+
+    UpdateCoord();
+    WallCyclic();
+    /*m_entities[i].xyz[0]+=m_entities[i].vxyz[0];
+        m_entities[i].xyz[1]+=m_entities[i].vxyz[1];
+        m_entities[i].xyz[2]+=m_entities[i].vxyz[2];*/
+
+    //COLLIDE WALL FUNCTION TO BUILD
+}
+
+float Core::BoidsGenerator::randomZeroOne()
+{
+    return static_cast<float>(rand()) / static_cast<float>(RAND_MAX);
+}
+
+float Core::BoidsGenerator::norm3D(float x, float y, float z)
+{
+    return sqrt(x * x + y * y + z * z);
+}
+
+void Core::BoidsGenerator::UpdateCoord()
```

Comment:
Please don't use Capital letter for function start, updateCoords() will do the job

Reply:
Be careful with looping around two data arrays in the same time, you can easily get a segfault if one of them has a size inferior to numEntities. 
Good practice to add a safety check like at the start : if(m_phy_boids.size() > m_numEntities || m_entities.size() > m_numEntities) return;